src/panels/config/backup/ha-config-backup-location.ts

```diff
@@ -0,0 +1,369 @@
+import { css, html, LitElement, nothing } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import "../../../components/ha-alert";
+import "../../../components/ha-button";
+import "../../../components/ha-switch";
+import "../../../components/ha-button-menu";
+import "../../../components/ha-card";
+import "../../../components/ha-circular-progress";
+import "../../../components/ha-icon-button";
+import "../../../components/ha-list-item";
+import "../../../components/ha-md-list";
+import "../../../components/ha-md-list-item";
+import type {
+  BackupAgent,
+  BackupAgentConfig,
+  BackupConfig,
+} from "../../../data/backup";
+import {
+  CLOUD_AGENT,
+  computeBackupAgentName,
+  fetchBackupAgentsInfo,
+  updateBackupConfig,
+} from "../../../data/backup";
+import "../../../layouts/hass-subpage";
+import type { HomeAssistant } from "../../../types";
+import "./components/ha-backup-data-picker";
+import { showConfirmationDialog } from "../../lovelace/custom-card-helpers";
+import { fireEvent } from "../../../common/dom/fire_event";
+
+@customElement("ha-config-backup-location")
+class HaConfigBackupDetails extends LitElement {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ type: Boolean }) public narrow = false;
+
+  @property({ attribute: "agent-id" }) public agentId!: string;
+
+  @property({ attribute: false }) public config?: BackupConfig;
+
+  @state() private _agent?: BackupAgent | null;
+
+  @state() private _agentIds?: string[];
+
+  @state() private _error?: string;
+
+  protected firstUpdated(changedProps) {
+    super.firstUpdated(changedProps);
+
+    if (this.agentId) {
```

Comment:
Shouldnt this be done on every update of `agentId` instead of on firstUpdated?

Reply:
The agent id can only be changed after a router change or a page refresh.

Reply:
Yeah, would still prefer to change the code so it also functions correct if we ever change the navigate actions, everything wil stay the same as it is right now if it doesn't change anyway 🤷‍♂️

Reply:
Done!